src/Tasks/WriteCodeFragment.cs

```diff
@@ -106,6 +106,11 @@ public override bool Execute()
 
             try
             {
+                if (OutputDirectory != null && !Path.IsPathRooted(OutputFile?.ItemSpec))
+                {
+                    FileUtilities.EnsureDirectoryExists(OutputDirectory.ItemSpec);
+                }
+
```

Comment:
My instinct is to move this to after `OutputFile` is guaranteed to be defined, so the task will still create the required folder(s) for a fully-specified `OutputFile` when there's no `OutputDirectory`. Sound reasonable?

Reply:
You mean where no `OutputDirectory` is provided and the `OutputFile` is a path and filename? Yes, I missed that case. There should be a unit test added to cover that.

Reply:
committed a change